### PR TITLE
Add reference solution to Lab 10 assignment

### PR DESCRIPTION
**Pull request by ivanangelov**

### CODE REVIEW

File: 10-network/lab/solution/src/bg/sofia/uni/fmi/mjt/wish/list/WishListServer.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        ServerSocketChannel sockChannel = (ServerSocketChannel) key.channel();
+        SocketChannel accept = sockChannel.accept();
+        accept.configureBlocking(false);
+        accept.register(selector, SelectionKey.OP_READ);
+
+        System.out.println("Connection accepted from client " + accept.getRemoteAddress());
+    }
+
+    private String postWish(String arguments) {
+        String[] argumentss = arguments.split("\\s+", 2);
```
Comment (100yo) [inline 10-network/lab/solution/src/bg/sofia/uni/fmi/mjt/wish/list/WishListServer.java:145]: Maybe rename "argumentss" to e.g. "argumentsArray"

**Review by 100yo — CHANGES_REQUESTED** [review]
Proposed some improvements, could you check them?

File: 10-network/lab/solution/src/bg/sofia/uni/fmi/mjt/wish/list/WishListServer.java (hunk 1 above)
Comment (ivanangelov) [reply 10-network/lab/solution/src/bg/sofia/uni/fmi/mjt/wish/list/WishListServer.java]: Done.

File: 10-network/lab/solution/src/bg/sofia/uni/fmi/mjt/wish/list/WishListServer.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+                    SelectionKey key = keyIterator.next();
+                    if (key.isReadable()) {
+                        SocketChannel socketChannel = (SocketChannel) key.channel();
+
+                        messageBuffer.clear();
+                        int r = socketChannel.read(messageBuffer);
+                        if (r <= 0) {
+                            System.out.println("Nothing to read, closing channel");
+                            socketChannel.close();
+                            break;
```
[the remark this answers is not in the document]
Comment (ivanangelov) [reply 10-network/lab/solution/src/bg/sofia/uni/fmi/mjt/wish/list/WishListServer.java]: Done.

File: 10-network/lab/solution/src/bg/sofia/uni/fmi/mjt/wish/list/WishListServer.java [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        try (ServerSocketChannel serverSocketChannel = ServerSocketChannel.open()) {
+            serverSocketChannel.bind(new InetSocketAddress(SERVER_HOST, port));
+            serverSocketChannel.configureBlocking(false);
+
+            Selector selector = Selector.open();
+            serverSocketChannel.register(selector, SelectionKey.OP_ACCEPT);
+
+            while (isStarted) {
+                int readyChannels = selector.select();
+                if (readyChannels == 0) {
```
[the remark this answers is not in the document]
Comment (ivanangelov) [reply 10-network/lab/solution/src/bg/sofia/uni/fmi/mjt/wish/list/WishListServer.java:52]: Done.

**Review by 100yo — APPROVED** [review]
Thanks for fixing my comments, it's good to go.

**Pull request merged**